client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -118,7 +119,16 @@ export default ( { attributes, className, setAttributes } ) => {
 								</Button>
 							</h4>
 							{ displayDate && (
-								<span className={ `${ className }__preview-post-date` }>{ post.date }</span>
+								<time
+									dateTime={ moment( post.date )
+										.utc()
+										.format() }
```

Comment:
I think you don't need `utc()` and rather than format, we should use [`toISOString()`](https://momentjs.com/docs/#/displaying/as-iso-string/).

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time#Valid_datetime_Values